### PR TITLE
Create CLAUDE.md for AI assistant guidance (#1138)

### PR DESCRIPTION
**Pull request by ddjain**
### **User description**
# Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [X] Optimization

# Description  
Added CLAUDE.md documentation for Krkn Chaos Engineering Framework, detailing project overview, repository structure, quick start guide, critical requirements, plugin architecture, testing procedures, cloud provider implementations, configuration, code style, exit codes, container support, git workflow, environment variables, common pitfalls, and development tasks.

## Related Tickets & Documents
#1138 
# Documentation  
- [ ] **Is documentation needed for this update?** -> NO

If checked, a documentation PR must be created and merged in the [website repository](https://github.com/krkn-chaos/website/).

## Related Documentation PR (if applicable)  
<-- Add the link to the corresponding documentation PR in the website repository -->  

# Checklist before requesting a review
[ ] Ensure the changes and proposed solution have been discussed in the relevant issue and have received acknowledgment from the community or maintainers. See [contributing guidelines](https://krkn-chaos.dev/docs/contribution-guidelines/)
See [testing your changes](https://krkn-chaos.dev/docs/developers-guide/testing-changes/) and run on any Kubernetes or OpenShift cluster to validate your changes
- [ ] I have performed a self-review of my code by running krkn and specific scenario 
- [ ] If it is a core feature, I have added thorough unit tests with above 80% coverage

*REQUIRED*:
Description of combination of tests performed and output of run

```bash
python run_kraken.py
...
<---insert test results output--->
```

OR

```bash
python -m coverage run -a -m unittest discover -s tests -v
...
<---insert test results output--->
```

___

### **PR Type**
Documentation

___

### **Description**
- Added comprehensive CLAUDE.md guidance document for AI assistants

- Covers project overview, repository structure, and quick start setup

- Details critical plugin architecture with strict naming conventions

- Includes testing procedures, cloud provider implementations, and configuration

- Provides git workflow, environment variables, common pitfalls, and development tasks

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Project Documentation"] --> B["CLAUDE.md"]
  B --> C["Project Overview & Structure"]
  B --> D["Plugin Architecture Rules"]
  B --> E["Testing & Development Guide"]
  B --> F["Cloud Provider Support"]
  B --> G["Git Workflow & Best Practices"]
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CLAUDE.md</strong><dd><code>Comprehensive AI assistant guidance for Krkn framework</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

CLAUDE.md

<ul><li>Created comprehensive AI assistant guidance document for Krkn Chaos <br>Engineering Framework<br> <li> Documented project overview, repository structure, and quick start <br>instructions<br> <li> Detailed critical plugin architecture with strict naming conventions <br>and implementation requirements<br> <li> Included testing procedures, unit test patterns, and functional test <br>guidelines<br> <li> Covered cloud provider implementations and configuration management<br> <li> Provided git workflow, environment variables, common pitfalls, and <br>development tasks</ul>

</details>

  </td>
  <td><a href="https://github.com/krkn-chaos/krkn/pull/1139/files#diff-6ebdb617a8104a7756d0cf36578ab01103dc9f07e4dc6feb751296b9c402faf7">+255/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by paigerube14** [comment]
@ddjain are you able to signoff your PR using steps: https://github.com/krkn-chaos/krkn/pull/1139/checks?check_run_id=61986163510

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+## Quick Start
+
+```bash
+# Setup (ALWAYS use virtual environment)
+python3 -m venv venv
+source venv/bin/activate
+pip install -r requirements.txt
+
+# Run Krkn
+python run_kraken.py --config config/config.yaml --scenario scenarios/openshift/pod.yml
```
Comment (paigerube14) [inline CLAUDE.md]: '--scenario' is not a proper command line argument. All scenario files need to be defined in the config

File: CLAUDE.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+```
+
+**Test requirements:**
+- Naming: `test_<module>_scenario_plugin.py`
+- Mock external dependencies (Kubernetes API, cloud providers)
+- Test success, failure, and edge cases
+- Keep tests isolated and independent
+
+### Functional Tests
+Located in `CI/tests/functional_tests/`. **DO NOT run locally** - requires full Kubernetes cluster, Prometheus, and Elasticsearch. Run automatically in CI.
```
Comment (paigerube14) [inline CLAUDE.md]: should be able to run locally on a kind cluster that has prometheus and elasticsearch set up. 

Set up can be found here: https://krkn-chaos.dev/docs/developers-guide/testing-changes/#prometheus and https://krkn-chaos.dev/docs/developers-guide/testing-changes/#prometheus 

Or can turn off prometheus and elastic using these parameters: https://github.com/krkn-chaos/krkn/blob/37ca4bbce750ca460b3657683ff938cdc17872e9/CI/config/common_test_config.yaml#L24-L25 and https://github.com/krkn-chaos/krkn/blob/37ca4bbce750ca460b3657683ff938cdc17872e9/CI/config/common_test_config.yaml#L28

File: CLAUDE.md (hunk 1 above)
Comment (ddjain) [reply CLAUDE.md]: Somehow this PR get closed without merge :/ , so created new pr with addressing comment [Link](https://github.com/krkn-chaos/krkn/pull/1141/changes/f26507dc4c82d4acbd56512549186f14f63f41be#diff-6ebdb617a8104a7756d0cf36578ab01103dc9f07e4dc6feb751296b9c402faf7R32)

File: CLAUDE.md (hunk 2 above)
Comment (ddjain) [reply CLAUDE.md]: Addressed comment at [Link](https://github.com/krkn-chaos/krkn/pull/1141/changes/f26507dc4c82d4acbd56512549186f14f63f41be#diff-6ebdb617a8104a7756d0cf36578ab01103dc9f07e4dc6feb751296b9c402faf7R116)
